thrust/testing/transform.cu

```diff
@@ -749,3 +749,38 @@ THRUST_DISABLE_BROKEN_GCC_VECTORIZER void TestTransformWithIndirection()
   ASSERT_EQUAL(output, ref);
 }
 DECLARE_INTEGRAL_VECTOR_UNITTEST(TestTransformWithIndirection);
+
+void TestTransformN()
+{
+  using namespace thrust::placeholders;
+
+  thrust::device_vector vec{1, 2, 3};
+  thrust::device_vector stencil{true, false, true};
+
+  {
+    thrust::device_vector result{0, 0, 0};
+    thrust::transform_n(vec.begin(), 3, result.begin(), _1 * 2);
```

Comment:
I would love to have a test that does not go the full length of the sequence to ensure that we are indeed stopping at the right position